### ISSUE TITLE
Page level error messages are missing, or not set up properly, or could be improved.

### ISSUE THREAD

**Issue by amazingphilippe** [cds-snc/notification-planning#1470]
## Description of issue

When pages are returned with errors, an overall error message that indicates that there is a problem with the form can be useful. Currently, the page reloads with the same heading, no overall error and only in-page errors. As a result, some users may not be aware of the errors until they go through the page content. 

For example, in the Create Service flow. When the service name and email address fields are returned with errors, users may not be aware of these errors until the navigate through the inputs a second time.
<img width="633" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMXVjQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a0b07c2c9ecd6b21443abaf54e65def642733b60/image.png" alt="image.png" />

The same technique used on the Sign in page to communicate when someone has entered an incorrect password could be used here; an error message is displayed in a <div> with the alert role and a tabindex of -1, and focus is moved to this <div> when the page is loaded. This immediately announces the error message, but keeps the <div> out of the keyboard focus order.

<img width="1335" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMTJjQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--5c098d15fd5d4e277a1f9e509c34a0ecaa57c672/image.png" alt="image.png" />

For more information on tabindex="-1", see MDN Web Docs tabindex page: 
Similarly status message do not have status roles. The status message observed on the Notify site also change the user’s context: they appear on a separate page, and the programmatic page title and main page heading also communicate the success status. These techniques communicate this information to the user and so the lack of a status role does not constitute a failure under 4.1.3 Status Messages. That being said, a role="status" with a tabindex of -1 and shifting of focus to the status message may still pose some benefit (for example by announcing the date and time the message was sent without requiring the user to navigate to the page content, as this information appears in the status). In addition, the status currently appears above the main page heading and when users skip to main content they bypass this message. We recommend testing this scenario with screen reader and braille display users to establish if a status role would provide additional benefit or whether the existing techniques in place are sufficient to provide clarity. 

E.g., the following status message that appears when a test email is sent does not have a status role. 
<img width="1179" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMkNjQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--6f390b8a44a9e3aed91da299815b1538836d7911/image.png" alt="image.png" />

List of screens to fix:
- [ ] Create service name
- [ ] Send test message success

## Potential fix

- Use the status banner component on test message success. 
- See if we have more screens that could use it. Probably by searching for the class: `banner-default-with-tick`. fix those too.
- Test different scenarios with Fable
- Add an error summary to the service creation flow step 2 (create service name and email)
- Remember to create error summary when screens have more than 1 input field

## Resources

- [MDN Web Docs tabindex](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex#sect2)

**Comment by andrewleith** [comment]
This is an interesting idea and certainly a good practice to follow.  I had some ideas on how this might work and tried it out on the create service page. By implementing it, it highlights to me that our form labels still need some work.  I had mentioned this during the "what is a service" work but it is more evident here:

![Image](https://github.com/cds-snc/notification-planning/assets/823749/498806a8-aff6-4854-8fb8-afd6729a3e8b)

Since the labelling isn't concise, it becomes confusing when you try to relate the error message to the field.  For example, instead of "Enter the part before ‘@notification.canada.ca’: this cannot be empty" it would be much clearer with concise labels like: "Email address: this cannot be empty"

**Comment by andrewleith** [comment]
Here are the changes I was suggesting for this form in particular during the "what is a service" work:

## Current
![Image](https://github.com/cds-snc/notification-planning/assets/823749/c8cfcf36-a55c-45bf-ad1b-9f634b130875)

## Recommendation
![Image](https://github.com/cds-snc/notification-planning/assets/823749/741ab387-ef74-48c6-a81e-a67de0d6cbb5)

**Comment by yaelberger-commits** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Notify-Planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/1470?planning-poker) with Zenhub @amazingphilippe @andrewleith

**Comment by yaelberger-commits** [comment]
Error summary appears in Staging when form fields are left empty at the top of create account page
<img width="812" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMEc0QkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--0b21cd28e0c043791e168ef4a59c27f6368d87c8/Screenshot%202024-05-06%20at%209.26.53%E2%80%AFAM.png" alt="Screenshot 2024-05-06 at 9.26.53 AM.png" />

**Comment by yaelberger-commits** [comment]
En Français, aussi

<img width="743" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBME80QkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--ef47bc7b75566e850f50b45553373af62614b167/Screenshot%202024-05-06%20at%209.28.42%E2%80%AFAM.png" alt="Screenshot 2024-05-06 at 9.28.42 AM.png" />
